api_test.go

```diff
@@ -1587,4 +1590,205 @@ var _ = Describe("RestHandler", func() {
+			req, err := http.NewRequest("GET", "/posts/1?fields[posts]=title&fields[users]=name", nil)
+			Expect(err).ToNot(HaveOccurred())
+			api.Handler().ServeHTTP(rec, req)
+			Expect(rec.Code).To(Equal(http.StatusOK))
+			Expect(rec.Body.Bytes()).To(MatchJSON(`
+				{"data": {
+					"id": "1",
+					"type": "posts",
+					"attributes": {
+						"title": "Nice Post"
+					},
+					"relationships": {
+						"author": {
+							"data": {
+								"id": "666",
+								"type": "users"
+							},
+							"links": {
+								"related": "/posts/1/author",
+								"self": "/posts/1/relationships/author"
+								}
+						},
+						"bananas": {
+							"data": [],
+							"links": {
+								"related": "/posts/1/bananas",
+								"self": "/posts/1/relationships/bananas"
+								}
+							},
+						"comments": {
+							"data": [],
+							"links": {
+								"related": "/posts/1/comments",
+								"self": "/posts/1/relationships/comments"
+							}
+						}
+					}
+				},
+				"included": [
+					{
+						"attributes": {
+							"name": "Tester"
+						},
+						"id": "666",
+						"type": "users"
+					}
+				]
+			}`))
+		})
+
+		It("FindAll: only returns requested post field for single post and includes", func() {
+			req, err := http.NewRequest("GET", "/posts?fields[posts]=title&fields[users]=name", nil)
+			Expect(err).ToNot(HaveOccurred())
+			api.Handler().ServeHTTP(rec, req)
+			Expect(rec.Code).To(Equal(http.StatusOK))
+			Expect(rec.Body.Bytes()).To(MatchJSON(`
+				{"data": [{
+					"id": "1",
+					"type": "posts",
+					"attributes": {
+						"title": "Nice Post"
+					},
+					"relationships": {
+						"author": {
+							"data": {
+								"id": "666",
+								"type": "users"
+							},
+							"links": {
+								"related": "/posts/1/author",
+								"self": "/posts/1/relationships/author"
+								}
+						},
+						"bananas": {
+							"data": [],
+							"links": {
+								"related": "/posts/1/bananas",
+								"self": "/posts/1/relationships/bananas"
+								}
+							},
+						"comments": {
+							"data": [],
+							"links": {
+								"related": "/posts/1/comments",
+								"self": "/posts/1/relationships/comments"
+							}
+						}
+					}
+				}],
+				"included": [
+					{
+						"attributes": {
+							"name": "Tester"
+						},
+						"id": "666",
+						"type": "users"
+					}
+				]
+			}`))
+		})
+
+		It("Summarize all invalid field query parameters as error", func() {
+			req, err := http.NewRequest("GET", "/posts?fields[posts]=title,nonexistent&fields[users]=name,title,fluffy,pink", nil)
+			Expect(err).ToNot(HaveOccurred())
+			api.Handler().ServeHTTP(rec, req)
+			Expect(rec.Code).To(Equal(http.StatusBadRequest))
+			error := HTTPError{}
+			err = json.Unmarshal(rec.Body.Bytes(), &error)
+			Expect(err).ToNot(HaveOccurred())
+
+			expectedError := func(field, objType string) Error {
+				return Error{
+					Status: "Bad Request",
+					Code:   "INVALID_FIELD_QUERY_PARAM",
```

Comment:
shouldn't the code namespaced with api2go? We can not know if a user already has a `INVALID_FIELD_QUERY_PARAM` :p